### PR TITLE
WIP: Merge mathematical logic and infrastructure improvements

### PR DESCRIPTION
**Pull request by vaibhavmehrotraml**
Primary Contributions:
- Change logic in acoustic_utils to match Ben's visualisations
- Parallelize data download and processing
- Change orca-hls-utils node to @vaibhavmehrotraml's implementation

WIP:
- Update documentation 
- Test edge cases
- Cost analysis

### CODE REVIEW

**Review by zprice12 — APPROVED** [review]
Looks good to me. We could try to implement the fastparquet write method if having RAM problems, but saw you added the delete for the lists of dataframes which should help with RAM

File: src/orcasound_noise/pipeline/acoustic_util.py [hunk 1]
```diff
@@ -217,21 +292,22 @@ def ancient_ambient(df):
 
 def spec_plot(df, sr=48000, hop_length=256):
```
Comment (valentina-s) [inline src/orcasound_noise/pipeline/acoustic_util.py:293]: There is a function `spec_plot` and `plot_spec`: very confusing.

File: src/orcasound_noise/pipeline/acoustic_util.py [hunk 2]
```diff
@@ -397,35 +474,87 @@ def plot_noise(testdf, name, output_path=None, save = False):
...
+def plot_bb(bb_df):
+    """
+    This function plots the broadband levels in relative decibels.
+
+    Args:
+        bb_df: Dataframe of broadband levels.
+
+    Returns: Time series of broadband levels.
+    """
+
```
Comment (valentina-s) [inline src/orcasound_noise/pipeline/acoustic_util.py:555]: add `plt.xticks(rotation = 45)` so that the xticks are visible

**Review by valentina-s — APPROVED** [review]
I have a few comments inline. Some of them became outdated with the new pushes. You can use the option 'Show outdated'. They are mainly about making the function names in quotes so that they are more readable. And a few small things. Thanks!

**Comment by valentina-s** [comment]
Thank you @zachprice12 @vaibhavmehrotraml and @ttan27 on your contributions!
